### PR TITLE
Add `inc-some-*` tests

### PR DESCRIPTION
**Pull request by shiatsumat**

### CODE REVIEW

**Comment by shiatsumat** [comment]
Because I wanted to use `rand::random` in tests, I let `creusot-contracts` re-export the `rand` crate. I believe this is convenient.

**Comment by xldenis** [comment]
That seems alright to me.

File: creusot/tests/should_succeed/inc_some_tree.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    Node(tl, a, tr) => sum_tree(*tl) + Int::from(a) + sum_tree(*tr),
+    Leaf => 0,
+  }
+  }
+}
+
+/* TODO: prove this lemma */
+#[trusted]
+#[ensures(sum_tree(*t) >= 0)]
+fn lemma_sum_tree_nonneg(t: &Tree) {}
```
Comment (xldenis) [inline creusot/tests/should_succeed/inc_some_tree.rs:26]: i think you should be able to prove these trees by doing induction in the function body. If you do 

```rust
fn lemma_sum_tree_nonneg(t: &Tree) {
match t {
Leaf => (),
Node(box l, n, box r) =>{ lemma_sum_tree_nonneg(l); lemma_sum_tree_nonneg(r) }
}
```

Comment (shiatsumat) [reply creusot/tests/should_succeed/inc_some_tree.rs:26]: Ah, I see.

Comment (shiatsumat) [reply creusot/tests/should_succeed/inc_some_tree.rs:26]: The problem is that this makes `lemma_sum_tree_nonneg` actually non-trivial as Rust code, taking unboundedly many program steps.
It is nice if we can mark lemma functions like `lemma_sum_tree_nonneg` as ghost code and eliminate them in the actual executable.

Comment (shiatsumat) [reply creusot/tests/should_succeed/inc_some_tree.rs:26]: Is this going to be supported in Creusot?

Comment (xldenis) [reply creusot/tests/should_succeed/inc_some_tree.rs:26]: >  It is nice if we can mark lemma functions like lemma_sum_tree_nonneg as ghost code and eliminate them in the actual executable.

Agreed. We (Claude, JH and me) discussed this this morning. I have a clear plan on how to add 'logic functions with contracts' which is what you need.

Comment (xldenis) [reply creusot/tests/should_succeed/inc_some_tree.rs:26]: The solution is actually what's outlined in #28, but we only use the logical part.

Comment (xldenis) [reply creusot/tests/should_succeed/inc_some_tree.rs:26]: I'd say give it 1-2 weeks for addition but if you write your "lemma"s in this style you would only need to add a `#[pure]` attribute to be able to use them in a _true_ ghost context.

Comment (shiatsumat) [reply creusot/tests/should_succeed/inc_some_tree.rs:26]: OK, then I'll write lemmas like that and just comment out the `#[pure]` annotations for now.

**Pull request merged**